### ISSUE TITLE
[BUG] tf.keras autolog does not log metric per epoch

### ISSUE THREAD

**Issue by edwardcqian** [mlflow/mlflow#4163]
### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [x] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- Ubuntu 20.04
- MLflow version: 1.12.0
- python version: 3.7.10

### Describe the problem
tf.keras autolog does not log metrics per epoch, only once.

### Code to reproduce issue
```
import tensorflow as tf
import mlflow


fashion_mnist = tf.keras.datasets.fashion_mnist

(train_images, train_labels), (test_images, test_labels) = fashion_mnist.load_data()
class_names = ['T-shirt/top', 'Trouser', 'Pullover', 'Dress', 'Coat',
               'Sandal', 'Shirt', 'Sneaker', 'Bag', 'Ankle boot']

model = tf.keras.Sequential([
    tf.keras.layers.Flatten(input_shape=(28, 28)),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dense(10)
])
model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

mlflow.tensorflow.autolog()
model.fit(train_images, train_labels, epochs=10)
```

### Other info / logs
Expected accuracy graph
<img width="917" alt="Screen Shot 2021-03-08 at 10 52 12 AM" src="https://user-images.githubusercontent.com/26368837/110345409-884a7800-7ffc-11eb-9d05-a2c11d553ae1.png">

Actual accuracy graph
<img width="918" alt="Screen Shot 2021-03-08 at 10 52 24 AM" src="https://user-images.githubusercontent.com/26368837/110345331-74067b00-7ffc-11eb-9b00-c2781f53c9b5.png">

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
Hi @edwardcqian, I think setting `every_n_iter` to 1 should solve the issue. 

```diff
+ mlflow.tensorflow.autolog()
- mlflow.tensorflow.autolog(every_n_iter=1)  # every_n_iter is 100 by default
```

**Comment by edwardcqian** [comment]
Ah, did not read the docs properly, thanks!
